app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
@phillipm, when you have a chance, let me know what your thoughts are based on that information. My inclination is that this is the correct thing to do, even if the code for it will be a bit messy. But would be good to know if you feel otherwise before I go ahead and do it.